### PR TITLE
Makes map view hiding a bit more robust against unusual circumstances

### PR DESCRIPTION
**Pull request by ninjanomnom**
While in the middle of a destroy chain the mob can no longer be associated with the client you need to clear huds from. This allows you to just pass a client into a different proc and has some more error checking to hopefully avoid visible issues for players if something goes wrong.

This fixes a runtime I cam across while working on something else.

### CODE REVIEW

File: code/_onclick/hud/map_view.dm [hunk 1]
```diff
@@ -55,15 +55,23 @@ INITIALIZE_IMMEDIATE(/atom/movable/screen/map_view)
 	return pop_planes
 
 /atom/movable/screen/map_view/proc/hide_from(mob/hide_from)
-	hide_from?.canon_client.clear_map(assigned_map)
-	var/client_ref = WEAKREF(hide_from?.canon_client)
+	var/client/target = hide_from?.canon_client
+	if(!target)
+		return
+
```
Comment (LemonInTheDark) [inline code/_onclick/hud/map_view.dm:61]: is this just reflexive? there's no reason to nullcheck here if we do so later

File: code/_onclick/hud/map_view.dm [hunk 2]
```diff
@@ -19,7 +19,7 @@ INITIALIZE_IMMEDIATE(/atom/movable/screen/map_view)
 		var/client/our_client = client_ref.resolve()
 		if(!our_client)
 			continue
```
Comment (LemonInTheDark) [inline code/_onclick/hud/map_view.dm:21]: similar

File: code/_onclick/hud/map_view.dm [hunk 3]
```diff
@@ -55,15 +55,23 @@ INITIALIZE_IMMEDIATE(/atom/movable/screen/map_view)
...
+	var/datum/weakref/client_ref = WEAKREF(hide_from)
 	// Make sure we clear the *right* hud
-	var/datum/weakref/hud_ref = viewers_to_huds[client_ref]
-	viewers_to_huds -= client_ref
-	var/datum/hud/clear_from = hud_ref?.resolve()
+	var/datum/hud/clear_from = viewers_to_huds[client_ref]?.resolve()
 	if(!clear_from)
 		return
 
+	viewers_to_huds -= client_ref
```
Comment (LemonInTheDark) [inline code/_onclick/hud/map_view.dm:75]: why is this being moved? surely we should remove the client ref even if clear_from fails to resolve

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/_onclick/hud/map_view.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/_onclick/hud/map_view.dm:61]: The way I previously had this organized it was stopping a potential runtime

**Comment by LemonInTheDark** [comment]
you mention an issue linked to like, not having ahold of the client at destroy. does this pr fix that or just make it less disruptive somehow

**Comment by ninjanomnom** [comment]
I don't think the client being dereferenced by the mob is itself an issue since it's happening during the destroy process, it's more that the map view cleanup needed to account for that being a possibility. So yeah this is fixing the issue as far as I'm aware.

**Pull request merged**